File: Code/Components/DanpheEMR.ServerModel/ClinicalModels/VitalsModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class VitalsModel
    {
        [Key]
        public int PatientVitalId { get; set; }
        public int PatientVisitId { get; set; }

        //assign default value to this property, don't set it from anywhere else.. 

        public double? Height { get; set; }
        public string HeightUnit { get; set; }
        public double? Weight { get; set; }
        public string WeightUnit { get; set; }
        public double? BMI { get; set; }
        public double? Temperature { get; set; }
        public string TemperatureUnit { get; set; }
        public int? Pulse { get; set; }
        public int? BPSystolic { get; set; }
        public int? BPDiastolic { get; set; }
        public string RespiratoryRatePerMin { get; set; }
        public double? SpO2 { get; set; }
        public string OxygenDeliveryMethod { get; set; }
        public int? PainScale { get; set; }
        public string BodyPart { get; set; }
        public VisitModel Visit { get; set; }
        public string Advice { get; set; }
        public string FreeNotes { get; set; }
        public string DiagnosisType { get; set; }
        public string Diagnosis { get; set; }
        public int? CreatedBy { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime VitalsTakenOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        //public virtual PatientModel Patient { get; set; }

        public int? Nadi { get; set; }
        public string Mala { get; set; }
        public string Mutra { get; set; }
        public string Jivha { get; set; }
        public string Shabda { get; set; }
        public string Sparsha { get; set; }
        public string Drik { get; set; }
        public string Akriti { get; set; }
        public string LungField { get; set; }
        public string HeartSounds { get; set; }
        public string PA_Tenderness { get; set; }
        public string Organomegaly { get; set; }
        public string CNS_Consiousness { get; set; }
        public string Power { get; set; }
        public string Reflexes { get; set; }
        public string Tone { get; set; }
        public string Others { get; set; }
    }
    }

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/clinical/shared/vitals.model.ts

//importing form and its related components, these are used for forms validation.
import {
    NgForm,
    FormGroup,
    FormControl,
    Validators,
    FormBuilder,
    ReactiveFormsModule
} from '@angular/forms'


export class Vitals {
    public PatientVitalId: number = 0;
    public PatientVisitId: number = 0;
    public DateTime: string = null;
    public Height: number = null;
    public HeightUnit: string = "cm";
    public Weight: number = null;
    public WeightUnit: string = "kg";
    
    //needs revision
    public BMI: number = null;
    public Temperature: number = null;
    public TemperatureUnit: string = "F";
    public Pulse: number = null;
    public BPSystolic: number = null;
    public BPDiastolic: number = null;
    public RespiratoryRatePerMin: string = null;
    public SpO2: number = null;
    public OxygenDeliveryMethod: string = null;
    public PainScale: number = null;
    public BodyPart: string = null;

    public CreatedBy: number = null;
    public ModifiedBy: number = null;
    public CreatedOn: string = null;
    public ModifiedOn: string = null;
    public VitalsTakenOn: string = null;

    public VitalsValidator: FormGroup = null;

    public  Nadi : string=null;
    public  Mala :string=null;
    public  Mutra :string=null
    public  Jivha :string=null
    public  Shabda :string=null
    public  Sparsha :string=null
    public  Drik :string=null
    public  Akriti :string=null
    public  LungField :string=null
    public  HeartSounds :string=null
    public  PA_Tenderness :string=null
    public  Organomegaly :string=null
    public  CNS_Consiousness :string=null
    public  Power :string=null
    public  Reflexes :string=null
    public  Tone :string=null
    public  Others:string=null

    constructor() {

        var _formBuilder = new FormBuilder();

        this.VitalsValidator = _formBuilder.group({
            'systolic': new FormControl(''),
            'diastolic': new FormControl(''),
        });

    }


    public IsDirty(): boolean {
        return this.VitalsValidator.dirty;
    }


    public IsValid():boolean{if(this.VitalsValidator.valid){return true;}else{return false;}} 
    public IsValidCheck(): boolean {
        let bpSys = this.VitalsValidator.controls["systolic"].value;
        let bpDias = this.VitalsValidator.controls["diastolic"].value;

        if (bpSys && bpDias) {
            if (bpSys < bpDias) {
                return false;
            }
            else if(bpSys == bpDias){
                return false;
            }
        }
        return true;
    }

    //returns true if both bp diastolic and systolic are filled.
    // false if one is filled and another is filled.
    public IsBPComplete(): boolean {

        let bpSys = this.VitalsValidator.controls["systolic"].value;
        let bpDias = this.VitalsValidator.controls["diastolic"].value;
        //if both empty then return true. 
        if (!bpSys && !bpDias) {
            return true;
        }
        else if (bpSys && !bpDias) {
            return false;
        }

        else if (bpDias && !bpSys) {
            return false;
        }
        return true;
    }
}
